code/modules/mob/living/carbon/carbon.dm

```diff
@@ -408,53 +408,75 @@
 		return 0
 	return ..()
 
-/mob/living/carbon/proc/vomit(lost_nutrition = 10, blood = FALSE, stun = TRUE, distance = 1, message = TRUE, vomit_type = VOMIT_TOXIC, harm = TRUE, force = FALSE, purge_ratio = 0.1)
+/// Proc that compels the mob to throw up. Returns TRUE if the mob actually threw up.
+/mob/living/carbon/proc/vomit(vomit_flags, lost_nutrition = 10, distance = 1, purge_ratio = 0.1)
+	if(vomit_flags == NONE)
+		stack_trace("Called vomit() on [src] with no vomit flags. Please add them in when you invoke this proc!")
+		return
```

Comment:
`vomit_flags` will never be `NONE` by default they're `null` 

But also consider, why are the default flags not just the default flags

```suggestion
/mob/living/carbon/proc/vomit(vomit_flags = VOMIT_CATEGORY_DEFAULT, lost_nutrition = 10, distance = 1, purge_ratio = 0.1)
```

Reply:
i considered doing it like this but i think it makes more sense for this to be a mandatory argument in the long run given the complex number of things you might want to do in a vomit

Reply:
I think there's merit to being able to just call `vomit()` and it just, does a vomit, you don't have to fudge around with looking for flags or whatever

Reply:
i suppose so, but i'll leave the invoking of the specific flag there anyways